src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -38,7 +38,20 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
             {
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
-                Console.Clear();
+
+                bool hasConsole = true;
+
+                try
+                {
+                    Console.Clear();
+                }
+                catch (IOException)
+                {
+                    // The console is not available, probably because it has been redirected.
+                    // See https://docs.microsoft.com/dotnet/api/system.console.clear#remarks
+                    hasConsole = false;
```

Comment:
While this works, I would prefer that we use the S.CommandLine way of checking for a console and/or preemptively checking if we have a console rather than catching the exception, e.g., 

This will only clear the terminal if we have a terminal
```csharp
console.GetTerminal()?.Clear(); // note this uses the IConsole console and not System.Console
```

This will check if we have a terminal to clear/rewrite the line in:
```csharp
if (console.GetTerminal() is ITerminal terminal)
{
    // use terminal here to do things like set cursor position
    // can also use System.Console here since we know we have one
}
```
or simply
```csharp
bool hasConsole = console.GetTerminal() != null;
```

Reply:
OK. I went with the last suggestion.